### ISSUE TITLE
[a11y]: useDatagrid() requires useless ascendingSortableLabelText, descendingSortableLabelText, defaultSortableLabelText props

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#5939]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.46.0

### React version

18

### Automated testing tool and ruleset

manual inspection

### Assistive technology

no but a blind user presumably would

### Description

According to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort and also https://w3c.github.io/aria/#aria-sort, the "aria-sort" parameter should be hardcoded to "ascending", "descending" or "none", regardless of the user's locale.

Therefore, useDatagrid() should not require the `ascendingSortableLabelText`, `descendingSortableLabelText`, and `defaultSortableLabelText` properties.

Concretely, there are two issues:

1. The application is required to specify those properties, or else Datagrid doesn't set `aria-sort`.
2. The application developer will likely be confused, and translate those properties, but that will likely break the screenreaders.

Side note: the current "Sortable columns" story is also broken because it mismatched the text to the property keys:

```
ascendingSortableLabelText: 'none',  <-- should have been "ascending"
descendingSortableLabelText: 'ascending', <-- should have. been "descending"
defaultSortableLabelText: 'descending', <-- should have been "default"
```

### WCAG 2.1 Violation

unclear

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-datagrid--sortable-columns&globals=viewport:basic

### Steps to reproduce

You could probably get an error by updating the above story to have translated names, and then test what the various screenreaders do.

But mainly this ticket is about removing unnecessary and confusing code.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by ljcarot** [comment]
We've been investigating [TanStack ](https://tanstack.com/table/latest)[Table](https://tanstack.com/table/latest), a third-party, open-source offering, which provides extensive data table capabilities surpassing what our Carbon Datagrid offers. It provides much more flexibility and customization. TanStack Table is headless which means it can easily be added alongside Datagrid component in your product or application. The benefits of more flexibility for product teams and less maintenance for Carbon makes it a win win. Lastly, it is available in multiple frameworks including [React](https://github.com/carbon-design-system/tanstack-carbon/tree/main/react#readme) and [Web Component](https://github.com/carbon-design-system/tanstack-carbon/tree/main/web-components#readme) so it provides an option to non-React product teams.

For these reasons, we have decided to transition from building our own custom table component to using an example-based approach with TanStack Table. Datagrid will still exist in our library for existing teams but we are announcing the deprecation* of the Datagrid component in v2.54.0 release so teams can begin to work through the transition. Details about how to use both Datagrid and TanStack together can be found [here](https://github.com/carbon-design-system/tanstack-carbon/tree/main). 

*Deprecation means that no new features will be added however sev 1 and sev 2 bugs will be supported.
